src/NServiceBus.Core/Transports/TransportReceiver.cs

```diff
@@ -3,24 +3,24 @@ namespace NServiceBus
     using System;
     using System.Threading.Tasks;
     using Logging;
-    using ObjectBuilder;
     using Transports;
 
     class TransportReceiver
     {
         public TransportReceiver(
             string id,
-            IBuilder builder,
+            CriticalError criticalError,
             PushSettings pushSettings,
             PushRuntimeSettings pushRuntimeSettings,
-            Func<IBuilder, PushContext, Task> onMessage)
+            IPipelineInvoker pipelineInvoker,
```

Comment:
this seems to touch with the recoverability changes, but I assume you have already checked that this won't have a significant impact?